llvm/test/SYCLLowerIR/PropagateAspectsUsage/PropagateFunctionWithSyclDetailAspect/propagate-aspects-from-function-with-sycl-detail-1.ll

```diff
@@ -0,0 +1,94 @@
+; RUN: opt -passes=sycl-propagate-aspects-usage %s -S | FileCheck %s
+
+; Check that baz() & bar() take the same !sycl_used_aspects as foo()
+
+; baz()
+;  |
+;  v
+; bar()
+;  |
+;  v
+; foo()
+
+source_filename = "main.cpp"
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-unknown"
+
+%class.anon = type { i8 }
+
+define weak_odr dso_local spir_kernel void @_ZTSZZ4mainENKUlRN4sycl3_V17handlerEE_clES2_EUlvE_() {
```

Comment:
We should ensure that the aspect usage metadata is propagated all the way up to the kernel.

Reply:
Done [a6da336](https://github.com/intel/llvm/pull/7035/commits/a6da336c7d130956827dc6a7c3c9a6fafd966aad)